addons/website/models/website_page.py

```diff
@@ -83,6 +90,14 @@ def _get_most_specific_pages(self):
             previous_page = page
         return self.browse(ids)
 
+    @api.constrains('parent_page_id')
+    def _check_parent_page_id(self):
+        for page in self:
+            if page.parent_page_id.id == page.id:
+                raise ValidationError(_("Page '%s' cannot be parent of itself") % page.name)
+            if page.is_homepage and page.parent_page_id:
+                raise ValidationError(_("Homepage '%s' cannot have a parent page") % page.name)
+
```

Comment:
Somehow this does not seem to work at all... if I follow this flow:
- Set the homepage as parent page of a "test" page
- Set the "test" page as parent page of a "child" page
-  Go to "test", set it as homepage

=> You now have an homepage with a breadcrumb and you cannot remove its parent because the field does not even show up.

cc @lebl-odoo

Reply:
Good catch. Indeed, the homepage shouldn't have a parent. If we set a page as the homepage, we should remove the parent field.